### PR TITLE
qa_crowbarsetup: cinder netapp multi-backend

### PR DESCRIPTION
**Pull request by MaximilianMeister**
enable simultaneous deployment of netapp nfs and iscsi storage protocol via:
* cinder_backend=netapp
* want_cindermultibackend=1

QA:
``` bash
export cinder_backend=netapp
export cinder_netapp_login=openstack
export cinder_netapp_password="[SECRET]"
export want_cindermultibackend=1
```

### CODE REVIEW

**Comment by jdsn** [comment]
:+1: lgtm

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -1586,16 +1586,23 @@ function dns_proposal_configuration()
...
-    proposal_set_value cinder default "${attr_volume_path}[0]['netapp']['storage_protocol']" "'${cinder_netapp_storage_protocol}'"
-    proposal_set_value cinder default "${attr_volume_path}[0]['netapp']['netapp_server_hostname']" "'netapp-n1-e0m.cloud.suse.de'"
-    proposal_set_value cinder default "${attr_volume_path}[0]['netapp']['vserver']" "'cloud-openstack-svm'"
-    proposal_set_value cinder default "${attr_volume_path}[0]['netapp']['netapp_login']" "'${cinder_netapp_login}'"
-    proposal_set_value cinder default "${attr_volume_path}[0]['netapp']['netapp_password']" "'${cinder_netapp_password}'"
-    if [ "$cinder_netapp_storage_protocol" = 'nfs' ] ; then
-        proposal_set_value cinder default "${attr_volume_path}[0]['netapp']['nfs_shares']" "'netapp-n1-nfs.cloud.suse.de:/n1_vol_openstack_nfs'"
+    local volnumber=$1
+    local p="proposal_set_value cinder default"
+    local a="['attributes']['cinder']['volumes']"
```
Comment (bmwiedemann) [inline scripts/qa_crowbarsetup.sh]: $a could include [$volnumber]

File: scripts/qa_crowbarsetup.sh [hunk 2]
```diff
@@ -1831,16 +1838,25 @@ function custom_configuration()
...
 
             # add a second backend to enable multi-backend, if not already present
             if [[ $want_cindermultibackend = 1 ]] ; then
-                if ! crowbar cinder proposal show default | grep -q local-multi; then
+                # in case of testing netapp, add a second backend with a different storage protocol
+                if [[ $cinder_backend = "netapp" ]]; then
+                    if [[ $cinder_netapp_storage_protocol = "iscsi" ]] ; then
+                        cinder_netapp_storage_protocol="nfs"
+                    else
+                        cinder_netapp_storage_protocol="iscsi"
```
Comment (bmwiedemann) [inline scripts/qa_crowbarsetup.sh]: might be better to have this value as explicit param to the cinder_netapp_proposal_configuration function

File: scripts/qa_crowbarsetup.sh [hunk 3]
```diff
@@ -1586,16 +1586,23 @@ function dns_proposal_configuration()
...
-    proposal_set_value cinder default "${attr_volume_path}[0]['netapp']['vserver']" "'cloud-openstack-svm'"
-    proposal_set_value cinder default "${attr_volume_path}[0]['netapp']['netapp_login']" "'${cinder_netapp_login}'"
-    proposal_set_value cinder default "${attr_volume_path}[0]['netapp']['netapp_password']" "'${cinder_netapp_password}'"
-    if [ "$cinder_netapp_storage_protocol" = 'nfs' ] ; then
-        proposal_set_value cinder default "${attr_volume_path}[0]['netapp']['nfs_shares']" "'netapp-n1-nfs.cloud.suse.de:/n1_vol_openstack_nfs'"
+    local volnumber=$1
+    local p="proposal_set_value cinder default"
+    local a="['attributes']['cinder']['volumes']"
+    if [[ $volnumber -gt 0 ]]; then
+        proposal_modify_value cinder default "$a" "{}" "<<"
```
Comment (MaximilianMeister) [inline scripts/qa_crowbarsetup.sh]: > $a could include [$volnumber]

but not here, that's why i left it out of `$a`

**Comment by bmwiedemann** [comment]
:+1: (did not test though)

**Pull request merged**